task 2/2 intro task.playground/Contents.swift

```diff
@@ -0,0 +1,200 @@
+        }
+        self.lastElement = newStackElement
+    }
+    
+    /**
+     Returns the top element of the stack, but does not remove it. Complexity: O(1)
+     - returns: The top element of the stack, but does not remove it.
+     */
+    public func top() -> Int? {
+        return lastElement?.data
+    }
+
+    /**
+     Removes and returns the top element of the stack. Complexity: O(1)
+     - returns: The top element of the stack or nil when the invoking stack is empty.
+     */
+    public func pop() -> Int? {
+        if self.firstElement == nil{
+            return nil
+        }
+        let resultData = self.lastElement!.data
+        if let previousElement = self.lastElement?.previousElement {
+            self.lastElement = previousElement
+        } else {
+            self.firstElement = nil
+            self.lastElement = nil
+        }
+        
+        return resultData
+    }
+}
+
+/**
+ Class with stack realisation and for getting statistics of stack, inheritance from class Stack()
+ */
+public class StackStatistics: Stack {
+    /**
+    Array of min elements of stack
+     */
+    private var minElements: [Int]
+
+    override public init() {
+        self.minElements = [Int]()
+        super.init()
+    }
+
+    /**
+    Input value compared with last element of minElements array  and if new value <= last array element, newvalue appended to minElement array.
+    - Complexity: O(1)
+    - parameter element: Input value for checking as min element of stack
+    */
+    private func minElementCheck(element: Int) {
+        if let minElement = self.minElements.last {
+            if element <= minElement {
+                self.minElements.append(element) // append has complexity O(1)
+            }
+        } else {
+            self.minElements.append(element)
+        }
+    }
+
+    /**
+     Add element on the top of the stack.
+     - Complexity: O(1)
+     - parameter element: Element for push to stack
+     */
+    override public func push(_ data: Int) {
+        self.minElementCheck(element: data)
+        super.push(data)
+    }
+
+    /**
+    Removes and returns the top element of the stack.
+    - Complexity: O(1)
+    - returns: The top element of the stack or nil when the stack is empty.
+    */
+    override public func pop() -> Int? {
+        if let lastItem = super.pop() {
+            if let minElement = self.minElements.last {
+                if lastItem == minElement {
+                    self.minElements.removeLast() //Complexity: O(1)
+                }
+            }
+            
+            return lastItem
+        }
+        
+        return nil
+    }
+    
+    /**
+    Returns the first minimum element in the stack.
+    - Complexity: O(1)
+    - returns: The first minimum element in the stack.
+    */
+    public func minimumElement() -> Int? {
+        return self.minElements.last  // Complexity: O(1)
+    }
+    
+}
+
+/**
+Print minimumElement and pop() elements of input stack
```

Comment:
```suggestion
 Print minimumElement and pop() elements of input stack
```

Please use the default indentation.

Reply:
Third time: I'm still able to see doc comments with non-default indentation.